### PR TITLE
docs(ctb): Document the testing conventions

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Documents the proposed conventions for organizing our tests.

### CODE REVIEW

**Comment by maurelian** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4122** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4122" target="_blank">Graphite</a>  👈
    * **PR #4124** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4124" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4122?utm_source=stack-comment).

File: packages/contracts-bedrock/.gas-snapshot
Comment (smartcontracts) [inline packages/contracts-bedrock/.gas-snapshot:8]: Gas snapshot change seems irrelevant here

**Review by smartcontracts — COMMENTED** [review]
LGTM other than the erroneous gas snapshot diff (?)

File: packages/contracts-bedrock/.gas-snapshot
Comment (maurelian) [reply packages/contracts-bedrock/.gas-snapshot]: oops, thanks

**Review by clabby — APPROVED** [review]
LGTM

**Comment by maurelian** [comment]
whoa... something went wrong with graphite. setting to draft.

**Review by mslipper — APPROVED** [review]

**Comment by mslipper** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/4122?utm_source=stack-comment).


Job Reference: zGQ1LoPqtyqGyH7GZuBd

**Pull request merged**
